### PR TITLE
Difference in $type handling for json.net btw embedded core v6 and the external serializer

### PR DESCRIPTION
**Pull request by andreasohlund**
This was discovered when @HEskandari was testing the https://docs.particular.net/samples/sqltransport/native-integration/?version=sqltransport_4 sample

### CODE REVIEW

File: nservicebus/serialization/newtonsoft.md [hunk 1]
```diff
@@ -62,7 +62,9 @@ snippet: NewtonsoftBson
 
 ## Compatibility with the core JSON serializer
 
-The only incompatibility with the [core serializer](json.md) is that this serializer does not support the serialization of `XContainer` and `XDocument` properties. If XML properties are required on messages strings should be used instead. If `XContainer` and `XDocument` properties are required [use a JsonConverter](newtonsoft.md#compatibility-with-the-core-json-serializer-use-a-jsonconverter-for-xcontainer-and-xdocument).
+### No support for `XContainer` and `XDocument` properties
+
+In contrast to the [core serializer](json.md) is that this serializer does not support the serialization of `XContainer` and `XDocument` properties. If XML properties are required on messages strings should be used instead. If `XContainer` and `XDocument` properties are required [use a JsonConverter](newtonsoft.md#compatibility-with-the-core-json-serializer-use-a-jsonconverter-for-xcontainer-and-xdocument).
```
Comment (timbussmann) [inline nservicebus/serialization/newtonsoft.md]: In contrast to the core serializer, this serializer [...] ?

File: nservicebus/serialization/newtonsoft.md [hunk 2]
```diff
@@ -76,18 +78,30 @@ The `AddDeserializer` API can help transition between serializers. See the [Mult
...
-#### Use the JsonConverter
+##### Use the JsonConverter
 
 At configuration time the JsonConverter can then be used with the following.
 
 snippet: UseConverter
+
+### Use of $type requires an assembly qualified name
+
+The [core serializer](json.md) is registering a custom serialization binder to not require the assembly name to be present when inferring message type from the special [`$type` property supported by json.net](https://www.newtonsoft.com/json/help/html/SerializeTypeNameHandling.htm). Not having to specify the assembly name can be useful to reduce coupling when using the serializer for native integration scenarios like [demonstrated in this sample](/samples/sqltransport/native-integration).
```
Comment (timbussmann) [inline nservicebus/serialization/newtonsoft.md]: how much sense does it make to mention the core serializer when talking about v7? Can we just describe the current behavior instead?

File: nservicebus/serialization/newtonsoft.md [hunk 3]
```diff
@@ -76,18 +78,30 @@ The `AddDeserializer` API can help transition between serializers. See the [Mult
...
+
+To make the serializer compatible with this behavior the following serialization binder can be used:
```
Comment (timbussmann) [inline nservicebus/serialization/newtonsoft.md]: would that make more sense in an upgrade guide instead?

**Comment by andreasohlund** [comment]
@timbussmann updated based on our discussion, please take another look?

**Comment by weralabaj** [comment]
Why approved but not merged? @timbussmann is it waiting for anything more?

**Comment by andreasohlund** [comment]
Seems to be some unrelated build failure?

**Comment by HEskandari** [comment]
Rebased.

**Comment by HEskandari** [comment]
@andreasohlund I've addressed some issues, but looks like a snippet is missing now?
The error is:

Failed to render article. PackageId:NServiceBus.Newtonsoft.Json. Version:[2.0.0-pre, 3.0.0).
MissingSnippet, Key: KnownTypesBinder

**Pull request merged**